### ISSUE TITLE
[python-package] How to print training dataset metrics?

### ISSUE THREAD

**Issue by yanrudan** [microsoft/LightGBM#6707]
I use the `from lightgbm import LGBMRegressor` and my code is:

```python
lightgbm = LGBMRegressor(
    boosting_type='gbdt',
    objective=objective,
    metric='l1',
    random_state=0,
    num_leaves=100,
    n_estimators=200
)

model = lightgbm.fit(train_x, train_y, train_w)
```

My operations are all performed in the terminal of VSCode. 

I add `is_provide_training_metric=True` in LGBMRegressor, but there is no output. 
What should I do?

lightgbm version: 4.5.0

**Comment by yanrudan** [comment]
Thanks！I tried for a long time this morning and found that there is no `is_provide_training_metric` parameter for the LGBMRegressor API; it can only be used with `callbacks`. 

I have another question:
I can set the `metric` parameter when initializing the LGBMRegressor model, and I can also set the `eval_metric` parameter in the `fit` method. After testing, I found that both of these are evaluation metrics for the `eval_set`, is that correct?

**Comment by jmoralez** [comment]
> I can set the metric parameter when initializing the LGBMRegressor model, and I can also set the eval_metric parameter in the fit method. After testing, I found that both of these are evaluation metrics for the eval_set, is that correct?

Yes. You can set them in either place and they end up in the `params` argument of [lightgbm.train](https://lightgbm.readthedocs.io/en/stable/pythonapi/lightgbm.train.html#lightgbm-train)

**Comment by yanrudan** [comment]
thank you very much！
